tests/unit/widget/backcompat_core.js

```diff
@@ -0,0 +1,274 @@
+( function( $ ) {
+
+// Convert the values of the keys under option classes to hashes so they may be compared via
+// deepEquals. We convert to hash because order does not matter in a hash, just as it does not
+// matter in a class string.
+function convertClassValuesToHashes( options ) {
+	var classKey, classArray, classIndex;
+
+	if ( options.classes ) {
+		for ( classKey in options.classes ) {
+			if ( typeof options.classes[ classKey ] === "string" ) {
+				classArray = options.classes[ classKey ] ?
+					options.classes[ classKey ].split( " " ) : [];
+
+				// Overwrite the string-typed class value with a hash
+				options.classes[ classKey ] = {};
+				for ( classIndex in classArray ) {
+					options.classes[ classKey ][ classArray[ classIndex ] ] = true;
+				}
+			}
+		}
+	}
+
+	return options;
+}
+
+module( "Synchronization of classes option and style options", {
+	setup: function() {
+		$.widget( "mobile.testwidget", {
+			options: {
+				classes: {
+					"test-testwidget": "ui-button-inline ui-shadow"
+				},
+				inline: true,
+				mini: false,
+				shadow: true,
+				corners: false
+			},
+			classProp: "test-testwidget",
+			_create: function() {
+				this._addClass( this.element, "test-testwidget" );
+				this._enhance();
+			},
+			_enhance: $.noop
+		} );
+		$.widget( "mobile.testwidget", $.mobile.testwidget, $.mobile.widget.backcompat );
+	},
+	teardown: function() {
+		delete $.mobile.testwidget;
+		delete $.fn.testwidget;
+	}
+} );
+
+test( "Style options are updated when the classes option changes", function( assert ) {
+	var testWidget = $( "<div>" );
+
+	testWidget.testwidget( {
+		classes: {
+			"test-testwidget": "ui-corner-all"
+		}
+	} );
```

Comment:
Just do `var testWidget = $.mobile.testWidget();` then you can do `testWidget.option( "classes", ...` and `testWidget.option() `, `testWidget.element[ 0 ]` etc below